contribute/dev_contribution.rst

```diff
@@ -0,0 +1,110 @@
+
+.. contents::
+
+Overview
+========
+
+For any body of work intended for the :term:`prime repository` start with a GitHub issue.  The issue can be used to discuss the topic and clarify any problems related to it.  Once progress has been made towards addressing the issue a pull request is created that references the issue.
+
+Reviewers provide feedback on the changes by adding comments to the pull request or associated commits. The Jenkin's build/test procedure will run each time changes are pushed to the pull request's branch, and the results are displayed in the pull request view.
+
+Once all the changes and reviews are complete, one of the :term:`prime repository` owners will merge the pull request into the prime repository, onto the develop branch.
+
+Note that a bug is just a type of issue, and that resolving the bug should have both the implementation to fix the bug and a test that triggers the bug.
+
+:numref:`Figure %s <OpenCMISS-development-process>` gives a graphical overview of the developer contribution process.  For more detail see the text below.
+
+.. _OpenCMISS-development-process:
+.. figure:: images/OpenCMISSProcesses-DevelopmentProcess.png
+   :align: center
+   :alt: Developer contribution process
+
+   Developer contribution process
+
+GitHub Issue
+============
+
+If an issue does not exist for the required work (e.g. implementation of a feature, fixing of a bug), then create a new one.  The issue is the place to discuss the particulars related to the issue, discussions on determining the scope of the issue or clarification of any points that are unclear.
+
+Labels
+------
+
+A GitHub Issue may be assigned labels by the project administrators to help identify its status at a glance. General labels currently used for OpenCMISS are:
+
+* **Bug**: The issue identifies a malfunction in the current codebase.
+* **Feature**: The issue constitutes a request or plan for a new feature.
+* **Needs tests**: The issue requires test(s) to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+* **Needs documentation**: The issue requires documentation to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+* **Needs reviewing**: The issue requires further review from project participants to be complete. This may refer to a bug report, contributed code, comments, etc. in the issue.
+
+In addition, a **Platform** label may be used to identify the issue as specific to a given platform (Windows/Linux/OS X). **Milestone** labels may be used to project when a feature is expected to be complete and/or indicate the priority of a given issue. Higher priority issues will take precedence and therefore be assigned a more immediate (lower) milestone number.
+
+Topic Branch
```

Comment:
Consistently call them feature branches, or topic branches if there is consensus. Github has excellent documentation and calls them feature branch; I much prefer feature branch over topic branch.